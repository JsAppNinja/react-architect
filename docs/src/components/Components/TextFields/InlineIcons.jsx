import React from 'react';
import Button from 'react-md/lib/Buttons/Button';
import FontIcon from 'react-md/lib/FontIcons';
import SVGIcon from 'react-md/lib/SVGIcons';
import TextField from 'react-md/lib/TextFields';

import removeRedEye from 'icons/remove_red_eye.svg';
import phone from 'icons/phone.svg';

import './_icon-styles.scss';

const InlineIcons = () => (
  <div className="md-grid text-fields">
    <div className="md-cell text-fields__blocked-phones">
      <TextField
        id="phone-number-with-icon-left"
        label="Phone"
        type="tel"
        leftIcon={<FontIcon>phone</FontIcon>}
        size={10}
        fullWidth={false}
      />
      <TextField
        id="phone-number-with-icon-right"
        label="Phone"
        type="tel"
        rightIcon={<FontIcon>phone</FontIcon>}
        size={10}
        fullWidth={false}
      />
    </div>
    <div className="md-cell text-fields__blocked-phones">
      <TextField
        id="phone-number-with-svg-icon-left"
        label="Phone"
        type="tel"
        leftIcon={<SVGIcon use={phone.url} />}
        size={10}
        fullWidth={false}
      />
      <TextField
        id="phone-number-with-svg-icon-right"
        label="Phone"
        type="tel"
        rightIcon={<SVGIcon use={phone.url} />}
        size={10}
        fullWidth={false}
      />
    </div>
    <TextField
      id="text-with-close-button"
      label="Some text"
      inlineIndicator={<Button icon className="text-fields__inline-btn">clear</Button>}
      className="md-cell md-cell--bottom"
    />
    <TextField
      id="text-field-with-svg-password"
      label="Password with SVG"
      type="password"
      passwordIcon={<SVGIcon use={removeRedEye.url} />}
      className="md-cell md-cell--bottom"
    />
  </div>
);
export default InlineIcons;
