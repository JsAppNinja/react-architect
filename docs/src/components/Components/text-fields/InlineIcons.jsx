import React from 'react';
import Button from 'react-md/lib/Buttons/Button';
import FontIcon from 'react-md/lib/FontIcons';
import TextField from 'react-md/lib/TextFields';

import './_icon-styles.scss';

const InlineIcons = () => (
  <div className="md-grid text-fields">
    <div className="md-cell text-fields__blocked-phones">
      <TextField
        id="phone-number-with-icon-left"
        label="Phone"
        type="tel"
        leftIcon={<FontIcon>phone</FontIcon>}
        size={10}
        fullWidth={false}
      />
      <TextField
        id="phone-number-with-icon-right"
        label="Phone"
        type="tel"
        rightIcon={<FontIcon>phone</FontIcon>}
        size={10}
        fullWidth={false}
      />
    </div>
    <TextField
      id="text-with-close-button"
      label="Some text"
      inlineIndicator={<Button icon className="text-fields__inline-btn">clear</Button>}
      className="md-cell md-cell--bottom"
    />
  </div>
);
export default InlineIcons;
