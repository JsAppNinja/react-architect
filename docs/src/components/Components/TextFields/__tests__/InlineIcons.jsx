/* eslint-env jest */
import React from 'react';
import { render } from 'enzyme';

import InlineIcons from '../InlineIcons';

describe('InlineIcons', () => {
  it('should render correctly', () => {
    const tree = render(<InlineIcons />);
    expect(tree).toMatchSnapshot();
  });
});
