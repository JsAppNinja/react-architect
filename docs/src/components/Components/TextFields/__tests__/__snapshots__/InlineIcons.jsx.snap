// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineIcons should render correctly 1`] = `
<div
  class="md-grid text-fields"
>
  <div
    class="md-cell text-fields__blocked-phones"
  >
    <div
      class="md-text-field-container md-inline-block md-text-field-container--input"
    >
      <label
        class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-floating-label--icon-offset md-text--secondary"
        for="phone-number-with-icon-left"
      >
        Phone
      </label>
      <div
        class="md-text-field-icon-container"
      >
        <i
          class="md-icon material-icons md-text-field-icon md-text-field-icon--positioned"
        >
          phone
        </i>
        <div
          class="md-text-field-divider-container"
        >
          <input
            class="md-text-field md-text-field--floating-margin md-text"
            id="phone-number-with-icon-left"
            size="10"
            type="tel"
          />
          <hr
            class="md-divider md-divider--text-field md-divider--expand-from-left"
          />
        </div>
      </div>
    </div>
    <div
      class="md-text-field-container md-inline-block md-text-field-container--input"
    >
      <label
        class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary"
        for="phone-number-with-icon-right"
      >
        Phone
      </label>
      <div
        class="md-text-field-icon-container"
      >
        <div
          class="md-text-field-divider-container"
        >
          <input
            class="md-text-field md-text-field--floating-margin md-text"
            id="phone-number-with-icon-right"
            size="10"
            type="tel"
          />
          <hr
            class="md-divider md-divider--text-field md-divider--expand-from-left"
          />
        </div>
        <i
          class="md-icon material-icons md-text-field-icon md-text-field-icon--positioned"
        >
          phone
        </i>
      </div>
    </div>
  </div>
  <div
    class="md-cell text-fields__blocked-phones"
  >
    <div
      class="md-text-field-container md-inline-block md-text-field-container--input"
    >
      <label
        class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-floating-label--icon-offset md-text--secondary"
        for="phone-number-with-svg-icon-left"
      >
        Phone
      </label>
      <div
        class="md-text-field-icon-container"
      >
        <svg
          class="md-icon md-text-field-icon md-text-field-icon--positioned"
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
        <div
          class="md-text-field-divider-container"
        >
          <input
            class="md-text-field md-text-field--floating-margin md-text"
            id="phone-number-with-svg-icon-left"
            size="10"
            type="tel"
          />
          <hr
            class="md-divider md-divider--text-field md-divider--expand-from-left"
          />
        </div>
      </div>
    </div>
    <div
      class="md-text-field-container md-inline-block md-text-field-container--input"
    >
      <label
        class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary"
        for="phone-number-with-svg-icon-right"
      >
        Phone
      </label>
      <div
        class="md-text-field-icon-container"
      >
        <div
          class="md-text-field-divider-container"
        >
          <input
            class="md-text-field md-text-field--floating-margin md-text"
            id="phone-number-with-svg-icon-right"
            size="10"
            type="tel"
          />
          <hr
            class="md-divider md-divider--text-field md-divider--expand-from-left"
          />
        </div>
        <svg
          class="md-icon md-text-field-icon md-text-field-icon--positioned"
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        />
      </div>
    </div>
  </div>
  <div
    class="md-text-field-container md-full-width md-text-field-container--input md-cell md-cell--bottom"
  >
    <label
      class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary"
      for="text-with-close-button"
    >
      Some text
    </label>
    <input
      class="md-text-field md-text-field--inline-indicator md-text-field--floating-margin md-full-width md-text"
      id="text-with-close-button"
      type="text"
    />
    <hr
      class="md-divider md-divider--text-field md-divider--expand-from-left"
    />
    <button
      class="md-btn md-btn--icon md-pointer--hover md-inline-block md-text-field-inline-indicator md-text-field-inline-indicator--floating text-fields__inline-btn"
      type="button"
    >
      <div
        class="md-ink-container"
      >
        <span
          aria-hidden="true"
        />
      </div>
      <i
        class="md-icon material-icons md-text--inherit"
      >
        clear
      </i>
    </button>
  </div>
  <div
    class="md-text-field-container md-full-width md-text-field-container--input md-cell md-cell--bottom"
  >
    <label
      class="md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary"
      for="text-field-with-svg-password"
    >
      Password with SVG
    </label>
    <input
      class="md-text-field md-text-field--inline-indicator md-text-field--floating-margin md-full-width md-text"
      id="text-field-with-svg-password"
      type="password"
    />
    <hr
      class="md-divider md-divider--text-field md-divider--expand-from-left"
    />
    <button
      class="md-text-field-inline-indicator md-password-btn md-pointer--hover md-text-field-inline-indicator--floating md-text--disabled"
      type="button"
    >
      <svg
        class="md-icon"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      />
    </button>
  </div>
</div>
`;
